### PR TITLE
ci: enable arm64/aarch64 wheel builds

### PR DESCRIPTION
**Pull request by nikaro**
I gave a try at enabling builds for these architectures but did not succeed for linux/aarch64. What seems to happen (i may be wrong) is that the ARM docker images used by multibuild cannot be run on Linux amd64 platforms (the one used by GitHub runners), even with `--platform linux/arm64` flag. Cf. https://github.com/nikaro/gensim/actions/runs/3933985826

I cannot allocate more time try to fix this so i will left this PR as draft here, if someone want to continue the work on it or take inspiration from it (or close it) feel free to do it.

May be rewriting the wheels build CI from scratch using [cibuildwheel](https://github.com/pypa/cibuildwheel) could be easier, i did not test but to me it looks cleaner that the mess of bash scripts that multibuild is.

Refs https://github.com/RaRe-Technologies/gensim/issues/2994 https://github.com/RaRe-Technologies/gensim/issues/3425 https://github.com/RaRe-Technologies/gensim/issues/3341

### CODE REVIEW

**Comment by piskvorky** [comment]
Thanks @nikaro , build-improving / maintenance tickets are always appreciated.

I'll leave this PR open in case you or someone else want to revisit it in the future.

**Comment by nikaro** [comment]
@piskvorky i finally gave a try to cibuildwheel and have something that seems to work, while being simpler. Can you take a look?

**Comment by nikaro** [comment]
By the way, since all your jobs run on these triggers:

```yaml
on:
  push:
    branches: [ develop ]
  pull_request:
    branches: [ develop ]
```

I suggest to use only one workflow file with `build_wheels` job running after `tests` job passed successfully. That would avoid to run `linters` before both build and tests. What do you think?

**Comment by nikaro** [comment]
Is there anything missing? I would like to be able to use this lib on ARM platforms with Python 3.11.

**Comment by piskvorky** [comment]
@niharo thanks! What's needed now is a review by @mpenkov .

In the meantime, you can use a "privately build" wheel, right? Or is the release blocking for you?

**Comment by nikaro** [comment]
> In the meantime, you can use a "privately build" wheel, right? Or is the merge / release somehow blocking for you?

Actually i'm not myself the "end-user" of it. It is used by some devs at my job, so it is in the dependencies of our app and thus shared by everyone. We are in the process of evaluating the upgrade to Python 3.11, and this lib is the last one preventing us make the jump 😅 But there is nothing urgent so i will wait for the merge and next release.

Thank you for your answer.

**Comment by piskvorky** [comment]
No problem. I appreciate the PR and consider chipping in at https://github.com/sponsors/piskvorky if you can (or your company). We're just a handful of volunteers with busy non-open-source jobs ourselves.

**Comment by piskvorky** [comment]
@mpenkov what's the status here? @nikaro why did you close the PR?

**Comment by nikaro** [comment]
Oh sorry, i think it closed automatically since i removed the fork from my repositories.

**Comment by mpenkov** [comment]
OK. Please give me some time to review this. This pretty much entirely changes the way we build wheels, so we need to make sure it works.

BTW, did you have a chance to test out the wheels built by your proposed changes?

File: .github/workflows/build-wheels.yml [hunk 1]
```diff
@@ -9,328 +9,59 @@ on:
...
+        uses: pypa/cibuildwheel@v2.12.0
+        env:
+          CIBW_ARCHS_LINUX: x86_64 aarch64
+          CIBW_ARCHS_MACOS: x86_64 arm64
+          CIBW_ARCHS_WINDOWS: AMD64 x86 ARM64
+          CIBW_BEFORE_BUILD: pip install numpy scipy
+          CIBW_SKIP: pp* cp36-* cp37-* *-win32 *_i686 *-musllinux_*
+          CIBW_TEST_COMMAND: pytest -rfxEXs --durations=20 --disable-warnings --showlocals --pyargs gensim
+          CIBW_TEST_REQUIRES: pytest testfixtures mock
+          CIBW_TEST_SKIP: cp38* cp39* cp310* *_aarch64 *_arm4 *_universal2
```
Comment (nikaro) [inline .github/workflows/build-wheels.yml:48]:
```suggestion
          CIBW_TEST_SKIP: cp38* cp39* cp310* *_aarch64 *_arm64 *_universal2
```

**Comment by nikaro** [comment]
> BTW, did you have a chance to test out the wheels built by your proposed changes?

I've been able to test the linux x86_64 wheel: `pip install wheelhouse/gensim-*.whl` and `import gensim` work fine (build with `cibuildwheel --only cp310-manylinux_x86_64` command).

**Comment by piskvorky** [comment]
@nikaro is there a step to automatically test each wheel after building it? The same way we test them now, in the current process. Thanks!

**Comment by nikaro** [comment]
@piskvorky this is already done, cf. https://github.com/RaRe-Technologies/gensim/actions/runs/4276229889/jobs/7444479965#step:4:467, excluding the ones listed in `CIBW_TEST_SKIP` env var.

**Comment by piskvorky** [comment]
@mpenkov over to you :)

File: .github/workflows/build-wheels.yml (hunk 1 above)
Comment (mpenkov) [reply .github/workflows/build-wheels.yml:48]: Can you please make this change @nikaro? I can't seem to do it from here.

**Comment by mpenkov** [comment]
@nikaro Where is the source code for this PR? Github is telling me that https://github.com/nikaro/gensim/tree/ci-qemu-cibuildwheel is missing.

**Comment by nikaro** [comment]
@mpenkov i removed my fork, that's why it was missing. I reinstated it you can check it out again now.

**Comment by mpenkov** [comment]
@nikaro Thanks! Can you please adjust this PR to allow maintainer commits? I need to fix a few things, e.g. the CIBW_TEST_SKIP variable, but I need your permission to do so.

File: .github/workflows/build-wheels.yml (hunk 1 above)
Comment (nikaro) [reply .github/workflows/build-wheels.yml:48]: It looks like that removing my fork messed up things a bit. I cannot commit my suggestion either. And pushes to my branch are not reflected here...

**Comment by nikaro** [comment]
@mpenkov i will close this PR and recreate a new one. The removal of my fork seems to have messed up things a bit.

**Pull request closed without merging**